tests/ert/unit_tests/resources/test_run_eclipse_simulator.py

```diff
@@ -15,148 +11,103 @@
     ):
-        # WHEN eclrun is run
-        erun.runEclipse(eclrun_config=eclrun_config)
-
-    # THEN the env provided to eclrun is the same
-    # as the env from ecl_config, but PATH has been
-    # prepended with the value from ecl_config
-    with open("env.json", encoding="utf-8") as f:
-        run_env = json.load(f)
-
-    expected_eclrun_env = eclrun_conf["eclrun_env"]
-    for key, value in expected_eclrun_env.items():
-        if value is None:
-            assert key not in run_env
-            continue  # Typically LSB_JOBID
-
-        if key == "PATH":
-            assert run_env[key].startswith(value)
-        else:
-            assert value == run_env[key]
+        monkeypatch.setenv(var, value)
+    yield
 
 
 @pytest.mark.integration_test
-@pytest.mark.usefixtures("use_tmpdir", "init_eclrun_config")
+@pytest.mark.usefixtures("use_tmpdir", "e100_env")
 @pytest.mark.requires_eclipse
 def test_ecl100_binary_can_produce_output(source_root):
     shutil.copy(
         source_root / "test-data/ert/eclipse/SPE1.DATA",
         "SPE1.DATA",
     )
-    econfig = ecl_config.Ecl100Config()
 
-    erun = ecl_run.EclRun("SPE1.DATA", None)
-    erun.runEclipse(eclrun_config=ecl_config.EclrunConfig(econfig, "2019.3"))
+    erun = run_reservoirsimulator.RunReservoirSimulator(
+        "eclipse", "2019.3", "SPE1.DATA"
+    )
+    erun.run_eclipseX00()
 
-    ok_path = Path(erun.runPath()) / f"{erun.baseName()}.OK"
-    prt_path = Path(erun.runPath()) / f"{erun.baseName()}.PRT"
+    ok_path = Path("SPE1.OK")
+    prt_path = Path("SPE1.PRT")
 
     assert ok_path.exists()
     assert prt_path.stat().st_size > 0
 
-    assert len(erun.parseErrors()) == 0
+    assert len(erun.parse_errors()) == 0
+
+    assert not Path("SPE1.h5").exists(), "HDF conversion should not be run by default"
 
 
 @pytest.mark.integration_test
+@pytest.mark.usefixtures("use_tmpdir", "e100_env")
 @pytest.mark.requires_eclipse
-@pytest.mark.usefixtures("use_tmpdir", "init_eclrun_config")
-def test_forward_model_cmd_line_api_works(source_root):
-    # ecl_run.run() is the forward model wrapper around ecl_run.runEclipse()
+def test_ecl100_binary_can_handle_extra_dots_in_casename(source_root):
+    """There is code dealing with file extensions in the Eclipse runner
+    so it better be tested to work as expected."""
+    shutil.copy(
+        source_root / "test-data/ert/eclipse/SPE1.DATA",
+        "SPE1.DOT.DATA",
+    )
+
+    erun = run_reservoirsimulator.RunReservoirSimulator(
+        "eclipse", "2019.3", "SPE1.DOT.DATA"
+    )
+    erun.run_eclipseX00()
+
+    ok_path = Path("SPE1.DOT.OK")
+    prt_path = Path("SPE1.DOT.PRT")
+
+    assert ok_path.exists()
+    assert prt_path.stat().st_size > 0
+    assert len(erun.parse_errors()) == 0
+
+
+@pytest.mark.integration_test
+@pytest.mark.requires_eclipse
+@pytest.mark.usefixtures("use_tmpdir", "e100_env")
+def test_runeclrun_argparse_api(source_root):
+    # Todo: avoid actually running Eclipse here, use a mock
```

Comment:
This test seems to do the same as `test_ecl100_binary_can_produce_output` at the moment.